### DISCUSSION TITLE
Accessing a "astro.renderToString" function

### DISCUSSION THREAD

**Discussion by reggi** [withastro/roadmap#254] [Proposal]
Moving from an issue I made here: https://github.com/withastro/astro/issues/4372

I'm interested in having a SSR based astro site with a dynamic endpoint that returns HTML instead of JSON. I'd like for the webpage to say submit some form data hit this endpoint, process the data and return HTML. I'd like to be able to render this HTML from an astro component and pass in props.

For this to work I'd the ability to render a astro component from within the api to a string. 

I'd want something similar to `react.renderToString`.

From what I understand there's the [Astro Compiler](https://github.com/withastro/compiler#transform-astro-to-valid-typescript) but that's more for converting an astro file to typescript AST, not rendering the HTML of an astro file, perhaps that's something i could use?

Here's an example:

```ts
import astro from 'astro'
import Example from './example.astro'

astro.renderToString(Example, { prop: true })
```

or

```ts
import astro from 'astro'
astro.renderToString('./example.astro', { prop: true })
```

I also did some searching and did find something like this but I'm not sure where that link went.

```ts
import Example from './example.astro'
Example.render({ prop: true )
```

**Comment by waspeer** [comment]
I had a similar issue. I opened a discussion in the RFC category for this here: https://github.com/withastro/rfcs/discussions/409

**Comment by florian-lefebvre** [comment]
Covered by #916
